File: libsignal-service/src/main/java/org/whispersystems/signalservice/internal/push/ReceiptCredentialRequestJson.java
package org.whispersystems.signalservice.internal.push;

import com.fasterxml.jackson.annotation.JsonProperty;

import org.signal.libsignal.zkgroup.receipts.ReceiptCredentialRequest;
import org.signal.core.util.Base64;

class ReceiptCredentialRequestJson {
  @JsonProperty("receiptCredentialRequest")
  private final String receiptCredentialRequest;

  ReceiptCredentialRequestJson(ReceiptCredentialRequest receiptCredentialRequest) {
    this.receiptCredentialRequest = Base64.encodeWithPadding(receiptCredentialRequest.serialize());
  }
}

File: lib/libsignal-service/src/main/java/org/whispersystems/signalservice/api/donations/ReceiptCredentialRequestJson.kt
/*
 * Copyright 2025 Signal Messenger, LLC
 * SPDX-License-Identifier: AGPL-3.0-only
 */

package org.whispersystems.signalservice.api.donations

import org.signal.core.util.Base64
import org.signal.libsignal.zkgroup.receipts.ReceiptCredentialRequest

internal class ReceiptCredentialRequestJson(
  val receiptCredentialRequest: String
) {
  constructor(receiptCredentialRequest: ReceiptCredentialRequest) : this(Base64.encodeWithPadding(receiptCredentialRequest.serialize()))
}
